### PR TITLE
Transform recursively, without JSON.parse()

### PR DESCRIPTION
**Pull request by dangodev**
Ditches the `JSON.parse()` approach which was limiting in 2 major ways:

- Couldn’t add `nullable` support (#112)
- Couldn’t add support for comments (now added! 🎉 )

At the end of the day, TypeScript isn’t JSON 🤷. And it turns out the original codebase was closer to what it needed to be. This PR refactors the output generation a little to be _even simpler_ than before.

### CODE REVIEW

File: src/v3.ts [hunk 1]
```diff
@@ -43,112 +38,98 @@ export default function generateTypesV3(
...
-      return node; // return by default
-    }
-  );
+      // 2. name (with “?” if optional property)
+      output += `"${key}"${!required || !required.includes(key) ? "?" : ""}: `;
+
+      // 3. open nullable
+      if (value.nullable) {
+        output += "(";
+      }
```
Comment (dangodev) [inline src/v3.ts:112]: cc @antonk52 sorry about doing this for you, but your PR pointed a flaw in the design. And I wanted to make sure I was solving for `nullable` (and it seems to be!).

So this _generates_, and it _looks_ correct, but I actually didn’t add tests in this PR making sure that `| null` works in all cases. Are nullable tests (and potentially fixing some edge cases) something you’re interested in writing in a followup PR? 👍 / 👎 ?

File: tests/bin/expected/petstore.ts [hunk 2]
```diff
@@ -5,7 +5,17 @@
...
   schemas: {
-    Order: { id?: number; petId?: number; quantity?: number; shipDate?: string; status?: 'placed' | 'approved' | 'delivered'; complete?: boolean }
+    Order: {
+      id?: number
+      petId?: number
+      quantity?: number
+      shipDate?: string
+      /**
+       * Order Status
+       */
```
Comment (dangodev) [inline tests/bin/expected/petstore.ts:15]: This PR adds comment support! 🎉

File: tests/v3/expected/stripe.ts [hunk 3]
```diff
@@ -5,106 +5,325 @@
...
+      /**
+       * The business type.
+       */
       business_type?:
-        | "company"
-        | "government_entity"
-        | "individual"
-        | "non_profit";
+        | ("company" | "government_entity" | "individual" | "non_profit")
+        | null;
```
Comment (dangodev) [inline tests/v3/expected/stripe.ts:28]: I _believe_ `nullable` has been added, but we can’t be sure before we write unit tests.

File: src/v3.ts (hunk 1 above)
Comment (antonk52) [reply src/v3.ts:112]: No worries man, whatever gets the job done =)

IMHO I think the existing cases covered the nullable fairly well. If you can point out a few extra test cases where the explicit test cases are needed I have no problem sending a pr later on

Comment (dangodev) [reply src/v3.ts:112]: Yeah, fair. Thanks for understanding.

Also, a new version is up on npm with the changes, so I’d love to see if the new `@next` version works for your schemas :)
